src/SamplesApp/SamplesApp.UnitTests.Shared/Controls/UnitTest/UnitTestsControl.cs

```diff
@@ -50,12 +52,26 @@ public UnitTestsControl()
 		private void OnRunTests(object sender, RoutedEventArgs e)
 		{
 			_cts = new CancellationTokenSource();
-			_runner = Task.Run(async () => await RunTests(_cts.Token));
+			var filter = testFilter.Text.Trim();
+			if (string.IsNullOrEmpty(filter))
+			{
+				filter = null;
+			}
+
+			testResults.Children.Clear();
+			_runner = Task.Run(async () => await RunTests(_cts.Token, filter));
 		}
 
-		private void ReportMessage(string message)
+		private void ReportMessage(string message, bool isRunning = true)
 		{
-			var t = Dispatcher.RunAsync(Windows.UI.Core.CoreDispatcherPriority.Normal, () => runStatus.Text = message);
+			void Setter()
+			{
+				runButton.IsEnabled = !isRunning;
+				runningState.Text = isRunning ? "Running" : "Finished";
+				runStatus.Text = message;
+			}
+
+			var t = Dispatcher.RunAsync(Windows.UI.Core.CoreDispatcherPriority.Normal, Setter);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 't'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5037645)